biojava-core/src/main/java/org/biojava/nbio/core/sequence/io/ABITrace.java

```diff
@@ -0,0 +1,568 @@
+     */
+    public ABITrace(File ABIFile) throws IOException {
+        byte[] bytes = null;
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        FileInputStream fis = new FileInputStream(ABIFile);
+        BufferedInputStream bis = new BufferedInputStream(fis);
+        int b;
+        while ((b = bis.read()) >= 0) {
+            baos.write(b);
+        }
+        bis.close(); fis.close(); baos.close();
+        bytes = baos.toByteArray();
+        initData(bytes);
+    }
+
+    /**
+     * The URL constructor opens an ABI file from any URL.
+     * @param ABIFile is a <code>java.net.URL</code> for an ABI trace file.
+     * @throws IOException if there is a problem reading from the URL.
+     * @throws IllegalArgumentException if the URL does not contain a valid ABI file.
+     */
+    public ABITrace( URL ABIFile ) throws IOException
+    {
+        byte[] bytes = null;
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        InputStream is = ABIFile.openStream();
+        BufferedInputStream bis = new BufferedInputStream(is);
+        int b;
+        while ((b = bis.read()) >= 0)
+        {
+            baos.write(b);
+        }
+        bis.close(); is.close(); baos.close();
+        bytes = baos.toByteArray();
+        initData(bytes);
+    }
+
+    /**
+     * The <code>byte[]</code> constructor parses an ABI file represented as a byte array.
+     *
+     * @throws IllegalArgumentException if the data does not represent a valid ABI file.
+     */
+    public ABITrace(byte[] ABIFileData) {
+        initData(ABIFileData);
+    }
+
+    /**
+     * Returns the length of the sequence (number of bases) in this trace.
+     */
+    public int getSequenceLength() {
+        return SeqLength;
+    }
+
+    /**
+     * Returns the length of the trace (number of x-coordinate points in the graph).
+     */
+    public int getTraceLength() {
+        return TraceLength;
+    }
+
+    /**
+     * Returns an <code>int[]</code> array that represents the basecalls - each int in the
+     * array corresponds to an x-coordinate point in the graph that is a peak (a base location).
+     */
+    public int[] getBasecalls() {
+        return Basecalls;
+    }
+
+    /**
+     * Returns an <code>int[]</code> array that represents the quality - each int in the
+     * array corresponds to an quality value 90-255) in the graph at a base location).
+     */
+    public int[] getQcalls() {
+        return Qcalls;
+    }
+
+    /**
+     * Returns the original programmatically determined (unedited) sequence as a <code>AbstractSequence<NucleotideCompound></code>.
+     */
+    public AbstractSequence<NucleotideCompound> getSequence() throws CompoundNotFoundException {
+        try {
+            DNASequenceCreator creator = new DNASequenceCreator(ABITracerCompoundSet.getABITracerCompoundSet());
+            return creator.getSequence(sequence, 0);
+        } catch (Exception e) {
```

Comment:
No need to catch, just let the `CompoundNotFoundException` go through.

Catching a general `Exception` here could cause undesired side effects in the error handling.

If you think the `CompoundNotFoundException` can't happen (other than because of a bug), then it's valid to catch it (but only that exception), log a warning and not throw it.